packages/config/index.js

```diff
@@ -4,34 +4,68 @@ var path = require('path');
 
 var root = require('pkg-dir').sync(process.cwd());
 
-var npmConfig = require('./lib/parse-env')('hops');
-if (!Object.keys(npmConfig).length) {
-  try {
-    require.resolve(path.join(root, 'package.json'));
-    npmConfig = require(path.join(root, 'package.json')).config.hops;
-  } catch (_) {}
+var explorer = require('cosmiconfig')('hops', {
+  rcExtensions: true,
+  stopDir: root,
+  sync: true,
+});
+
+function getRawConfig() {
+  var result = explorer.load(process.cwd());
+  return Object.assign(
+    {
+      https: false,
+      host: '0.0.0.0',
+      port: 8080,
+      locations: [],
+      basePath: '',
+      assetPath: '',
+      browsers: '> 1%, last 2 versions, Firefox ESR',
+      node: 'current',
+      envVars: { HOPS_MODE: 'dynamic' },
+      moduleDirs: [],
+      appDir: '.',
+      buildDir: 'build',
+      cacheDir: 'node_modules/.cache/hops',
+    },
+    result ? result.config : {}
+  );
 }
 
-function extendConfig(config) {
-  if (npmConfig.extends) {
+function applyExtension(config) {
```

Comment:
As it is a breaking change anyway, we could think of getting rid of this mechanics anyway and propagate only one way of configuration (user can choose between js, yaml or package.json anyway).

Other possibility: extend this to an eslint like structure and allow several additional configs as an array. That way, a user config (like `~/hopslocal.js`) would be possible again.

Reply:
I would not want to get rid of that mechanism altogether as that would force basically everyone to add a JS config file to their projects - at least internally.

On the other hand, it appears to me that inheriting from multiple config modules is a rather advanced use-case. It such cases, I would consider it acceptable to have a JS config. Do you agree?

Reply:
You're right. Let's keep it simple...